internal/handlers/sjson/sjson.go

```diff
@@ -218,7 +217,8 @@ func Unmarshal(data []byte) (*types.Document, error) {
 		)
 	}
 
-	d := must.NotFail(types.NewDocument())
+	// TODO(quasilyte): MakeDocument(len(sch.Keys))?
```

Comment:
Yeah, seems reasonable. If the document is possible, it will have `sch.Keys` fields.
Same above.